plugins/kubernetes/app/models/kubernetes/release.rb

```diff
@@ -38,62 +39,21 @@ def nested_error_messages
       errors.full_messages + release_docs.flat_map(&:nested_error_messages)
     end
 
-    def watch_pods(&block)
-      @pod_watcher = Watchers::PodWatcher.new(client, namespace,
-                                              label_selector: pod_labels.to_kuber_selector,
-                                              log: true)
-      @pod_watcher.start_watching(&block)
-    end
-
-    def watch_rcs(&block)
-      @rc_watcher = Watchers::ReplicationControllerWatcher.new(client, namespace,
-                                                               label_selector: pod_labels.to_kuber_selector,
-                                                               log: true)
-
-      @rc_watcher.start_watching(&block)
-    end
-
-    def watch_pod_events(&block)
-      pod_names = @pod_watcher.try(:pod_names).presence || find_pod_names
-
-      if pod_names.empty?
-        Kubernetes::Util.log 'No pods to watch', release_id: id
-        return
-      end
-
-      @event_watcher = Watchers::EventWatcher.new(client, namespace,
-                                                  object_kind: 'Pod',
-                                                  object_names: pod_names,
-                                                  log: true)
-      @event_watcher.start_watching(&block)
-    end
-
-    def find_pod_names
-      if @pod_watcher
-        @pod_watcher.pod_names
-      else
-        pods = client.get_pods(namespace: namespace, label_selector: pod_labels.to_kuber_selector)
-        pods.map { |p| p.metadata.name }
-      end
-    end
-
-    def stop_watching
-      [@pod_watcher, @rc_watcher].each do |watcher|
-        watcher.stop_watching if watcher
-      end
-    end
-
     def docs_by_role
       @docs_by_role ||= release_docs.each_with_object({}) do |rel_doc, hash|
         hash[rel_doc.kubernetes_role.label_name] = rel_doc
       end
     end
 
+    def watch
+      Watchers::DeployWatcher.new(self).async :watch
```

Comment:
This is ok for now, but if we ever want to have `DeployWatcher`s supervised, `watch` would need to be called from the initialisation code. Otherwise when this actor is restarted by a supervisor, `watch` will not be omitted. I'm struggling with a similar issue myself ATM.